torch/_library/fake_impl.py

```diff
@@ -13,62 +13,76 @@ class FakeImplHolder:
 
     def __init__(self, qualname: str):
         self.qualname: str = qualname
-        self.kernel: Optional[Kernel] = None
-        self.lib: Optional[torch.library.Library] = None
-
-    def register(self, func: Callable, source: str) -> RegistrationHandle:
+        # kernels stores all registered fake kernels, ordered by registration
+        # time ascendingly (newer registration after older registration). If an
+        # operator library gets loaded that overrides an existing fake kernel,
+        # both kernels will be in the list, but the newest one will be the one
+        # that is run. If the library is unloaded, we will remove the kernel
+        # from this list.
+        self.kernels: list[Kernel] = []
```

Comment:
this aligns the implementation a little bit better with [OperatorEntry](https://fburl.com/code/gbjj4ol8) except I did this in ascending order instead of descending order bc prepending a list is a little more costly than appending

Reply:
When a library gets unloaded how is it determined which kernel to remove?

Reply:
nvm I figured it out (there's an O(N) traversal! but that's probably fine)